samples/cellular/modem_shell/README.rst

```diff
@@ -1148,13 +1160,13 @@ After programming the development kit, test it in the Linux environment by perfo
 Application FOTA support
 ========================
 
-To build the MoSh sample with application FOTA support, use the ``-DEXTRA_CONF_FILE=overlay-app_fota.conf`` option.
+To build the MoSh sample with application FOTA support, use the ``-DEXTRA_CONF_FILE=overlay-app_fota.conf`` and ``-DSB_CONFIG_BOOTLOADER_MCUBOOT=y`` options.
```

Comment:
although the docs seems correct, then I wonder, why haven't this sample been adjust to have the following files instead:
- `prj_fota.conf` (merged content of `prj.conf` + `overlay-app_fota.conf`)
- `sysbuild_fota.conf`

then a `sysbuild_fota.conf` containing `SB_CONFIG_BOOTLOADER_MCUBOOT=y` can be made, and users could simply build the sample with fota support by doing:
```
$ west build -p -b *board_target* -- -DFILE_SUFFIX=fota
```

and similar goes for the other updated and complex examples in this PR.

Reply:
> although the docs seems correct, then I wonder, why haven't this sample been adjust to have the following files instead:

Because we haven't known about `FILE_SUFFIX`. Seems it's not too old thing in Zephyr?
There are probably dozen(s) samples which could utilize this more so some info sharing would be nice on best practices in setting sample/app overlays (both Kconfig and DTC).

> * `prj_fota.conf` (merged content of `prj.conf` + `overlay-app_fota.conf`)

Copying a larger `prj.conf` with 50+ Kconfig options into all overlay config files doesn't sound too good. I guess there are no other possibilities to utilize `FILE_SUFFIX` than copying `prj.conf` into the desired `prj_<FILE_SUFFIX>.conf`?

Reply:
Yes, otherwise that looks nice, but considering the size of `modem_shell` `prj.conf`, we probably don't want to duplicate it. We would lose the benefits of an overlay file, i.e. only having to include those options which are configured differently.

Reply:
We decided that it's not feasible to use this mechanism for `prj.conf` and using it only for sysbuild options would probably not make the configuration clearer.